src/main/java/org/folio/service/LoanEventsGenerationService.java

```diff
@@ -60,14 +64,16 @@ protected Future<Loan> generateEvents(Loan loan) {
   }
 
   private Future<String> generateItemCheckedOutEvent(Loan loan) {
-   return checkedOutEventHandler.handleSkippingUserSummaryUpdate(new ItemCheckedOutEvent()
+    log.debug("generateItemCheckedOutEvent:: parameters loan: {}", () -> logAsJson(loan));
+    return checkedOutEventHandler.handleSkippingUserSummaryUpdate(new ItemCheckedOutEvent()
       .withLoanId(loan.getId())
       .withUserId(loan.getUserId())
       .withDueDate(loan.getDueDate())
       .withMetadata(loan.getMetadata()));
   }
 
   private Future<String> generateClaimedReturnedEvent(Loan loan) {
+    log.debug("generateClaimedReturnedEvent:: parameters loan: {}", () -> logAsJson(loan));
     if (CLAIMED_RETURNED_STATUS.equalsIgnoreCase(loan.getItemStatus())) {
```

Comment:
log condition or result?

Reply:
Added logging of the `if` condition here and also to `generateDeclaredLostEvent` and `generateDueDateChangedEvent` but I don't think it makes sense to log the result here. It's logged in `handleSkippingUserSummaryUpdate` already, logging it here won't add to much information. If it fails to update we'll see it anyway but if it succeeds we'll see two identical log records